client/src/actions/update-monument.js

```diff
@@ -0,0 +1,85 @@
+import {
+    FETCH_MONUMENT_UPDATE_PENDING, FETCH_MONUMENT_UPDATE_SUCCESS, FETCH_MONUMENT_UPDATE_ERROR, UPDATE_MONUMENT_PENDING,
+    UPDATE_MONUMENT_SUCCESS, UPDATE_MONUMENT_ERROR
+} from '../constants';
+import { addError } from './errors';
+import { get, put } from '../utils/api-util';
+
+function fetchMonumentForUpdatePending() {
+    return {
+        type: FETCH_MONUMENT_UPDATE_PENDING
+    };
+}
+
+function fetchMonumentForUpdateSuccess(monument) {
+    return {
+        type: FETCH_MONUMENT_UPDATE_SUCCESS,
+        payload: monument
+    };
+}
+
+function fetchMonumentForUpdateError(error) {
+    return {
+        type: FETCH_MONUMENT_UPDATE_ERROR,
+        error: error
+    };
+}
+
+function updateMonumentPending() {
+    return {
+        type: UPDATE_MONUMENT_PENDING
+    };
+}
+
+function updateMonumentSuccess(updatedMonument) {
+    return {
+        type: UPDATE_MONUMENT_SUCCESS,
+        payload: updatedMonument
+    };
+}
+
+function updateMonumentError(error) {
+    return {
+        type: UPDATE_MONUMENT_ERROR,
+        error: error
+    };
+}
```

Comment:
I'll do that, but there's not a nice way to abstract out the functions into `doAction` like you did unfortunately since the actions use different HTTP methods. It'd end up being more complex than good.